packages/next/src/server/future/route-modules/app-page/module.ts

```diff
@@ -22,6 +22,13 @@ if (process.env.NEXT_RUNTIME !== 'edge') {
   vendoredReactSSR = require('./vendored/ssr/entrypoints')
 }
 
+/**
+ * The AppPageModule is the type of the module exported by the bundled app page
+ * module.
+ */
+export type AppPageModule =
```

Comment:
Because the template files are compleyly valid TS, each module can in fact be used as the type base for each of the modules 😎